tests/interaction/line_edge_intersection.cc

```diff
@@ -0,0 +1,85 @@
+#include <fiddle/interaction/interaction_utilities.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_nothing.h>
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+
+#include <ibtk/AppInitializer.h>
+#include <ibtk/IBTKInit.h>
+
+using namespace dealii;
+using namespace SAMRAI;
+
+// Test the line edge intersection code
+
+void
+test(SAMRAI::tbox::Pointer<IBTK::AppInitializer> app_initializer)
+{
+  double d_stencil_center_x = 0.0;
+  double d_stencil_center_y = 0.0;
+  double d_circle_center_x  = 0.0;
+  double d_circle_center_y  = 0.0;
+  int    d_stencil_axis     = 0;
+  double d_circle_radius    = 1.0;
+  double d_stencil_width    = 1.0;
+  auto   input_db           = app_initializer->getInputDatabase();
+  auto   test_db            = input_db->getDatabase("test");
+  if (test_db->isDouble("stencil_center_x_coordinate"))
+    d_stencil_center_x = test_db->getDouble("stencil_center_x_coordinate");
+  if (test_db->isDouble("stencil_center_y_coordinate"))
+    d_stencil_center_y = test_db->getDouble("stencil_center_y_coordinate");
+  if (test_db->isDouble("circle_center_x_coordinate"))
+    d_circle_center_x = test_db->getDouble("circle_center_x_coordinate");
+  if (test_db->isDouble("circle_center_y_coordinate"))
+    d_circle_center_y = test_db->getDouble("circle_center_y_coordinate");
+  if (test_db->isDouble("stencil_width"))
+    d_stencil_width = test_db->getDouble("stencil_width");
+  if (test_db->isInteger("stencil_axis"))
+    d_stencil_axis = test_db->isInteger("stencil_axis");
+  if (test_db->isDouble("circle_radius"))
+    d_circle_radius = test_db->getDouble("circle_radius");
+  Triangulation<2>    tria_bulk;
+  Triangulation<1, 2> tria;
+  GridGenerator::hyper_ball(tria_bulk);
+  GridGenerator::hyper_sphere(tria,
+                              Point<2>(d_circle_center_x, d_circle_center_y),
+                              d_circle_radius);
+  tria.refine_global(6);
+  dealii::Point<2>     r(d_stencil_center_x, d_stencil_center_y);
```

Comment:
```suggestion
  const Point<2> r(test_db->getDoubleWithDefault("stencil_center_x_coordinate", 0.0),
                   test_db->getDoubleWithDefault("stencil_center_x_coordinate", 0.0));
```
We don't need all these extra variables.